### PR TITLE
Add tado X support

### PR DESCRIPTION
**Pull request by karlbeecken**
This adds basic tado X support.

The module auto-detects if it connects to a tado X account and uses other API endpoints if applicable.

Note: the API scheme of tado X differs severely sometimes, so with this patch old applications that want to use tado X may be adapted to also reflect these changes, some endpoints do not work the same way.

### CODE REVIEW

**Comment by Moritz-Schmidt** [comment]
Hey @wmalgadey , @palazzem , I would love to help maintain this project. I could help with the code quality improvements and typing you've started with in other PRs, I'd also like to improve the unit tests. I'm not really a pro, but I'm quite experienced with Python, especially with typing. Let me know if any help would be appreciated!

**Comment by wmalgadey** [comment]
From a professional point of view, I would recommend solving the problem with a new abstract base class ‘BaseRequest’ and two overloads ‘Tado’ and ‘TadoX’. But I know that this is a more complex solution. Should we create an issue for this new (maybe better) solution? Or am I being too picky?

File: PyTado/interface.py [hunk 1]
```diff
@@ -718,6 +815,10 @@ def set_open_window(self, zone):
         Note: This can only be set if an open window was detected in this zone
         """
 
+        if sef.http.isX:
```
Comment (wmalgadey) [inline PyTado/interface.py:818]:
```suggestion
        if self.http.isX:
```

**Comment by wmalgadey** [comment]
> Hey @wmalgadey , @palazzem , I would love to help maintain this project. I could help with the code quality improvements and typing you've started with in other PRs, I'd also like to improve the unit tests. I'm not really a pro, but I'm quite experienced with Python, especially with typing. Let me know if any help would be appreciated!

sure, every help is welcome 👍

**Comment by Moritz-Schmidt** [comment]
> From a professional point of view, I would recommend solving the problem with a new abstract base class ‘BaseRequest’ and two overloads ‘Tado’ and ‘TadoX’. But I know that this is a more complex solution. Should we create an issue for this new (maybe better) solution? Or am I being too picky?

Thanks for the fast response! Your suggestion sound good, I had a similar idea for the Zone class too. I'd be happy to implement the BaseRequest ABC and a Zone ABC in other PRs, but I would need some time with that. To get the Home Assistant Integration working with TadoX soon I'd appreciate it if you could merge this PR as is and create a new release.

**Comment by wmalgadey** [comment]
@Moritz-Schmidt take a look at https://github.com/wmalgadey/PyTado/pull/84, @palazzem already did quite a good effort in refactoring. I didn't merge it till now, because nobody responded, and I couldn't test the changes.

**Comment by Moritz-Schmidt** [comment]
@wmalgadey yes, I've already seen it and will probably continue the Work done there!

**Comment by wmalgadey** [comment]
When we merge this PR, there will be some conflicts in #84. Should we still merge?

**Comment by Moritz-Schmidt** [comment]
Yes I think we should merge, I can resolve the conflicts with #84 when I'm working on it. If we get this shipped in time, maybe we'll still get into HA 2024.11

**Comment by karlbeecken** [comment]
@wmalgadey Thank you very much for the fast merge!

**Comment by karlbeecken** [comment]
Ah @wmalgadey, could you also create a new release and push it to PyPI?

**Comment by wmalgadey** [comment]
done!

**Pull request merged**